todos-app/src/App/Auth/AuthForm.tsx

```diff
@@ -0,0 +1,185 @@
+import { useEffect, useState } from "react";
+import {
+  createUserWithEmailAndPassword,
+  onAuthStateChanged,
+  signInWithEmailAndPassword,
+  User,
+} from "firebase/auth";
+import { auth } from "../Firebase/FirebaseConfig";
+import "../css/Auth.css";
+
+type Props = {
+  updateUser(user: User | null): void;
+  user: User | null;
+};
+
+const AuthForm = ({ updateUser, user }: Props) => {
+  const [registerEmail, setRegisterEmail] = useState<string>("");
+  const [registerPassword, setRegisterPassword] = useState<string>("");
+  const [registerConfirmPassword, setRegisterConfirmPassword] =
+    useState<string>("");
+  const [loginEmail, setLoginEmail] = useState<string>("");
+  const [loginPassword, setLoginPassword] = useState<string>("");
+
+  //initialize auth state
+  useEffect(() => {
+    changeAuthState();
+  }, []);
+
+  const changeAuthState = () => {
+    onAuthStateChanged(auth, (currentUser) => {
+      updateUser(auth.currentUser);
+    });
+  };
+
+  const register = async (event: React.MouseEvent<HTMLElement>) => {
+    event.preventDefault();
+    if (registerPassword === registerConfirmPassword) {
+      try {
+        const user = await createUserWithEmailAndPassword(
+          auth,
+          registerEmail,
+          registerPassword
+        );
+
+        changeAuthState();
```

Comment:
are you sure this is necessary, looks like you already have an observer added when the form is loaded